### PR TITLE
Implement dictionary list and creation on graphql

### PR DESCRIPTION
**Pull request by progsmile**
#3354

### CODE REVIEW

File: apps/core/lib/core/utils/types_converter.ex [hunk 1]
```diff
@@ -14,4 +14,6 @@ defmodule Core.Utils.TypesConverter do
 
   def string_to_integer(string) when is_binary(string), do: String.to_integer(string)
   def string_to_integer(string), do: string
+
+  def keys_to_strings(%{} = map), do: map |> Jason.encode!() |> Jason.decode!()
```
Comment (kiurchv) [inline apps/core/lib/core/utils/types_converter.ex]: 😱

File: apps/core/test/support/factories.ex [hunk 2]
```diff
@@ -44,8 +44,8 @@ defmodule Core.Factories do
     |> repo_insert!(type)
   end
 
-  def insert_list(number_of_records, type, factory, attrs \\ []) do
-    for i <- 0..number_of_records, i > 0, do: insert(type, factory, attrs)
+  def insert_list(count, type, factory, attrs \\ []) do
+    for _ <- 1..count, do: insert(type, factory, attrs)
```
Comment (kiurchv) [inline apps/core/test/support/factories.ex]: How many records will be inserted if we pass 0 as the argument?

File: apps/graphql/test/web/resolvers/dictionary_resolver_test.exs [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+  end
+
+  defp call_list_dictionary(conn, variables) do
+    resp_body =
+      conn
+      |> post_query(@list_query, variables)
+      |> json_response(200)
+
+    resp_entities = get_in(resp_body, ~w(data dictionaries nodes))
+    paging = get_in(resp_body, ~w(data dictionaries pageInfo))
```
Comment (kiurchv) [inline apps/graphql/test/web/resolvers/dictionary_resolver_test.exs]: Let's use full names for the things

```suggestion
    page_info = get_in(resp_body, ~w(data dictionaries pageInfo))
```

File: apps/core/lib/core/utils/types_converter.ex (hunk 1 above)
Comment (progsmile) [reply apps/core/lib/core/utils/types_converter.ex]: Better ideas?

File: apps/core/test/support/factories.ex (hunk 2 above)
Comment (progsmile) [reply apps/core/test/support/factories.ex]: 1. it's incorrect, thanks

File: apps/graphql/test/web/resolvers/dictionary_resolver_test.exs (hunk 3 above)
Comment (progsmile) [reply apps/graphql/test/web/resolvers/dictionary_resolver_test.exs]: ok

File: apps/graphql/lib/graphql_web/resolvers/dictionary_resolver.ex [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    query
+    |> where([d], ilike(d.name, ^"%#{value}%"))
+    |> prepare_where(tail)
+  end
+
+  defp prepare_where(query, [{field, value} | tail]) do
+    query
+    |> where([l], field(l, ^field) == ^value)
+    |> prepare_where(tail)
+  end
```
Comment (PavelVesnin) [inline apps/graphql/lib/graphql_web/resolvers/dictionary_resolver.ex]: `prepare_where` has the same functionality that was already implemented in Search helper. E.g. [fragment](https://github.com/edenlabllc/ehealth.api/blob/develop/apps/graphql/lib/graphql_web/resolvers/helpers/search.ex#L51-L55)
Please use `Search` helper instead. You can see how it should be [here](https://github.com/edenlabllc/ehealth.api/blob/develop/apps/graphql/lib/graphql_web/resolvers/legal_entity_resolver.ex#L19-L35)

File: apps/graphql/test/test_helper.exs [hunk 5]
```diff
@@ -1 +1,2 @@
 ExUnit.start()
+ExUnit.configure(exclude: [pending: true])
```
Comment (PavelVesnin) [inline apps/graphql/test/test_helper.exs]: 👍

File: apps/graphql/test/web/resolvers/dictionary_resolver_test.exs [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+      insert_list(3, :il, :dictionary, is_active: false, labels: ["SOME_OPTION", "INTERNAL"])
+      insert(:il, :dictionary, name: "TEST_DICTIONARY")
+
+      :ok
+    end
+
+    test "list active items", %{conn: conn} do
+      {resp_body, resp_entities, paging} = call_list_dictionary(conn, %{first: 5, filter: %{is_active: true}})
+
+      assert nil == resp_body["errors"]
```
Comment (PavelVesnin) [inline apps/graphql/test/web/resolvers/dictionary_resolver_test.exs]: use [refute](https://hexdocs.pm/ex_unit/ExUnit.Assertions.html#refute/1)

File: apps/graphql/test/web/resolvers/dictionary_resolver_test.exs [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+      assert 5 == length(resp_entities)
+      assert %{"hasNextPage" => true, "hasPreviousPage" => false} = paging
+    end
+
+    test "list by name", %{conn: conn} do
+      {resp_body, resp_entities, paging} = call_list_dictionary(conn, %{first: 10, filter: %{name: "est_dict"}})
+
+      assert nil == resp_body["errors"]
```
Comment (PavelVesnin) [inline apps/graphql/test/web/resolvers/dictionary_resolver_test.exs]: use [refute](https://hexdocs.pm/ex_unit/ExUnit.Assertions.html#refute/1)

File: apps/graphql/test/web/resolvers/dictionary_resolver_test.exs [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+      assert [%{"name" => "TEST_DICTIONARY"}] = resp_entities
+      assert %{"hasNextPage" => false, "hasPreviousPage" => false} = paging
+    end
+
+    test "list inactive with label", %{conn: conn} do
+      {resp_body, resp_entities, paging} =
+        call_list_dictionary(conn, %{first: 5, filter: %{is_active: false, label: "INTERNAL"}})
+
+      assert nil == resp_body["errors"]
```
Comment (PavelVesnin) [inline apps/graphql/test/web/resolvers/dictionary_resolver_test.exs]: use [refute](https://hexdocs.pm/ex_unit/ExUnit.Assertions.html#refute/1)

File: apps/graphql/test/web/resolvers/dictionary_resolver_test.exs [hunk 9]
```diff
@@ -0,0 +1,151 @@
...
+      assert 3 == length(resp_entities)
+      assert %{"hasNextPage" => false, "hasPreviousPage" => false} = paging
+    end
+
+    test "not found by label search", %{conn: conn} do
+      {resp_body, resp_entities, _} = call_list_dictionary(conn, %{first: 10, filter: %{label: "NON_EXISTENT_LABEL"}})
+
+      assert nil == resp_body["errors"]
```
Comment (PavelVesnin) [inline apps/graphql/test/web/resolvers/dictionary_resolver_test.exs]: use [refute](https://hexdocs.pm/ex_unit/ExUnit.Assertions.html#refute/1)

File: apps/graphql/test/web/resolvers/dictionary_resolver_test.exs [hunk 10]
```diff
@@ -0,0 +1,151 @@
...
+      dictionary_params = %{
+        name: dictionary_name,
+        is_active: true,
+        labels: labels,
+        values: Jason.encode!(values)
+      }
+
+      {resp_body, resp_entity} = call_update_dictionary(conn, dictionary_params)
+
+      assert nil == resp_body["errors"]
```
Comment (PavelVesnin) [inline apps/graphql/test/web/resolvers/dictionary_resolver_test.exs]: use [refute](https://hexdocs.pm/ex_unit/ExUnit.Assertions.html#refute/1)

File: apps/graphql/test/web/resolvers/dictionary_resolver_test.exs [hunk 11]
```diff
@@ -0,0 +1,151 @@
...
+      insert(:il, :dictionary, name: dictionary_name, is_active: true)
+
+      dictionary_params = %{
+        name: dictionary_name,
+        is_active: false
+      }
+
+      {resp_body, resp_entity} = call_update_dictionary(conn, dictionary_params)
+
+      assert nil == resp_body["errors"]
```
Comment (PavelVesnin) [inline apps/graphql/test/web/resolvers/dictionary_resolver_test.exs]: use [refute](https://hexdocs.pm/ex_unit/ExUnit.Assertions.html#refute/1)

File: apps/graphql/test/web/resolvers/dictionary_resolver_test.exs (hunk 6 above)
Comment (progsmile) [reply apps/graphql/test/web/resolvers/dictionary_resolver_test.exs]: for me it's better to understand when assertion statement is positive
@kiurchv your thoughts

Comment (PavelVesnin) [reply apps/graphql/test/web/resolvers/dictionary_resolver_test.exs]: `assert` is for positive and
`refute` is for negative cases

File: apps/core/priv/repo/migrations/20181106160111_add_id_to_dictionaries.exs [hunk 12]
```diff
@@ -0,0 +1,9 @@
+defmodule Core.Repo.Migrations.AddIdToDictionaries do
+  use Ecto.Migration
+
+  def change do
+    alter table(:dictionaries) do
+      add(:id, :uuid)
```
[the remark this answers is not in the document]
Comment (progsmile) [reply apps/core/priv/repo/migrations/20181106160111_add_id_to_dictionaries.exs]: for relay pagination

**Pull request merged**